src/ro/onlinelibrary/library/OnlineLibrary.java

```diff
@@ -0,0 +1,218 @@
+        while(i<3){
+            if(books.get(i).getTitle() == null) {
+                booksBorrowedByAdults.get(adultId).set(i,book);
+                break;
+            }
+            else {
+                i++;
+            }
+        }
+    }
+
+// method to show info about a certain student/adult borrowings
+    public String showBorrowedBooks(String firstName, String lastName, int max, List<Books> booksBorrowed){
+        StringBuilder buffer = new StringBuilder();
+        buffer.append("<< " + firstName + " " + lastName + " >> \n");
+        buffer.append("\t Books borrow status:\n");
+        int counter = 0;
+        for (Books book: booksBorrowed) {
+            var title = book.getTitle();
+            if(title != null)
+            {
+                counter++;
+                buffer.append("\t " + counter + ". " + title + "("+book.getPublishingYear()+")\n");
+            }
+        }
+        buffer.append("\t Total: " + counter + " of " +max+ " possible.\n");
+        if(counter == 0){
+            buffer.append("\t No books borrowed at the moment!\n");
+        }
+        return buffer.toString();
+    }
+
+// method to show all the subscribed readers of the library
+    public String showReaders(){
+        StringBuilder buffer = new StringBuilder();
+        buffer.append("<< Student readers >>\n");
+        int counter  = 0;
+        for (Student student: students) {
+            counter++;
+            buffer.append(counter + ". " + student.getFirstName() + " " + student.getLastName() + "\n");
+            buffer.append("\t- School:" + student.getSchool() + "\n");
+            buffer.append("\t- Year:" + student.getYear() + "\n");
+        }
+        counter  = 0;
+        buffer.append("<< Adult readers >>\n");
+        for (Adult adult: adults) {
+            counter++;
+            buffer.append(counter + ". " + adult.getFirstName() + " "+ adult.getLastName() + "\n");
+            buffer.append("\t- Company:" + adult.getEmployerCompany()+ "\n");
+            buffer.append("\t- Job:" + adult.getJob() + "\n");
+        }
+        return buffer.toString();
+    }
+
+// method to show info about the books
+    public String showBooks(){
+        StringBuilder buffer = new StringBuilder();
+        buffer.append("<< Books >>\n");
+        int counter  = 0;
+        for (Books book: books) {
+            counter++;
+            buffer.append(counter + ". " + book.getTitle().toUpperCase() + " ("+book.getPublishingYear() +") by "+ book.getAuthor().getFirstName()
+                    + " " + book.getAuthor().getLastName() +"\n");
+            buffer.append("\t - Genre:" + book.getGenre() + "\n");
+            buffer.append("\t - Description:" + book.getDescription() + "\n");
+        }
+        return buffer.toString();
+    }
+
+    public static void sort(List<Books> books)
+    {
+        books.sort((book1, book2)
+                -> book1.getTitle().compareTo(
+                book2.getTitle()));
+    }

```

Comment:
redundant implementation of comparator, you already have it in NameComparator, just instantiate it and use it as param for sort here